core/src/main/java/com/linecorp/armeria/common/logging/RequestScopedMdc.java

```diff
@@ -231,7 +230,9 @@ public static void put(RequestContext ctx, String key, @Nullable String value) {
                 tmp.put(key, value);
                 newMap = Object2ObjectMaps.unmodifiable(tmp);
             }
-            ctx.setAttr(MAP, newMap);
+            if (ctx.setAttr(MAP, newMap) == oldMap) {
```

Comment:
This won't work. Let's say there are two threads trying to set attributes.
The thread `foo` tries to set `a=b` and the thread `bar` tries to set `c=d`.
Both threads are at line 233 and the thread `foo` calls `setAttr` first and then the thread `bar`.
What is the result?
The `a=b` will be completed disappeared because the thread `bar` cannot see the `a=b` from the ctx's map.

To solve this we need to add a lock inside of the `ctx` and use it. But I think that's probably overkill because this isn't heavy computation so we can just use the `synchroized` block.
That being said, how about reverting the changes in `RequestScopedMdc` class?

Reply:
Please forget about this comment. It works like a charm.

Reply:
>  That being said, how about reverting the changes in RequestScopedMdc class?

Sorry, I was confused. @minwoox's comment https://github.com/line/armeria/pull/4578/files#r1145790462 is correct. `ctx.setAttr(MAP, newMap)` updates the value anyway, we can't use compare and set approach. Let's revert it and use `synchroized` block.
We may exclude this from checkstyle lint later.

Reply:
@minwoox got it! reverted!